samples/linker-instructions-advanced.md

```diff
@@ -0,0 +1,63 @@
+# Using IL Linker Advanced Features
+
+This document described the more advanced features for the IL Linker and provides more insight into how it functions.
+
+The basic features and instructions for the linker are described in the [Using the .NET IL Linker](linker-instructions.md) document.
+
+## How the linker works
+
+At a high level, the linker uses a mark-and-sweep algorithm to remove unused code: starting from some set of roots in the code (roots can be classes, methods, properties), the linker scans the IL using mono/cecil to look for code that gets called in other functions, classes, and dlls, marking the code that is reachable from the set of roots as it goes. In the sweep phase, the linker will scan through all of the code, removing any parts that weren't marked in the mark phase.
+
+The dynamic features of .NET make it hard for this analysis to catch all cases in which code is called from the set of roots. In particular, reflection enables developers to do things like scan for assemblies at runtime and call their code. In general, it is impossible for the linker to determine exactly which code will get called at runtime, so there will be cases in which the linker removes code that should not be removed. We may add more sophisticated heuristics in the future to catch some common patterns of reflection usage, but the general problem remains, which necessitates a mechanism by which developers can explicitly tell the linker which parts of the code to consider as roots. Thus the linker operates on a set of assemblies and a set of specified code roots, producing an output set of assemblies with unused dependencies removed.
+
+## Using the tasks package
+
+The linker tasks package ([ILLink.Tasks](https://dotnet.myget.org/feed/dotnet-core/package/nuget/Illink.Tasks)) contains MSBuild targets that become a part of the referencing project's build, using the mechanism described in the nuget docs.
```

Comment:
Good idea. [This](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#including-msbuild-props-and-targets-in-a-package) would be the link.